tests/Microsoft.Identity.Test.Unit/CacheTests/TokenCacheTests.cs

```diff
@@ -44,6 +44,45 @@ public override void TestInitialize()
             base.TestInitialize();
         }
 
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public async Task WithAdalCacheCompatibilityTest_Async(bool enableAdalCacheCompatibility)
+        {
+            // Arrange
+            var legacyCachePersistence = Substitute.For<ILegacyCachePersistence>();
+            var serviceBundle = TestCommon.CreateServiceBundleWithCustomHttpManager(null, isAdalCacheEnabled: enableAdalCacheCompatibility);
+            var requestContext = new RequestContext(serviceBundle, Guid.NewGuid());
+            var response = TestConstants.CreateMsalTokenResponse();
+
+            ITokenCacheInternal cache = new TokenCache(serviceBundle, false);
+            ((TokenCache)cache).LegacyCachePersistence = legacyCachePersistence;      
+            
+            var requestParams = TestCommon.CreateAuthenticationRequestParameters(serviceBundle);
+            requestParams.TenantUpdatedCanonicalAuthority = Authority.CreateAuthorityWithTenant(
+                requestParams.AuthorityInfo,
+                TestConstants.Utid);
+            requestParams.Account = new Account(TestConstants.s_userIdentifier, $"1{TestConstants.DisplayableId}", TestConstants.ProductionPrefNetworkEnvironment);
+
+            // Act
+            await cache.FindRefreshTokenAsync(requestParams).ConfigureAwait(true);
+            await cache.SaveTokenResponseAsync(requestParams, response).ConfigureAwait(true);
+            await cache.GetAccountsAsync(requestParams).ConfigureAwait(true);
+            await cache.RemoveAccountAsync(requestParams.Account, requestContext).ConfigureAwait(true);
```

Comment:
Should I separate these each into their own test method?

Reply:
I suppose it is fine as is... to me it would depend on what you are asserting... this works for me as it's closely related.

---
In reply to: [551840387](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2309#discussion_r551840387) [](ancestors = 551840387)

Reply:
Thanks for clarifying.

The basic assertion is if the legacy cache is disabled, then cache operations don't call legacy cache methods; if legacy cache is enabled, normal cache operations end up calling the legacy cache methods.